### ISSUE TITLE
Bikeshedding

### ISSUE THREAD

**Issue by headinthebox** [ReactiveX/RxScala#81]
```
def apply[T](f: Subscriber[T] => Unit):        Observable[T]
def create[T](s: Observer[T] => Subscription): Observable[T]
```

`create` should also use `f`.

**Comment by samuelgruetter** [comment]
Changing

``` scala
def create[T](s: Observer[T] => Subscription): Observable[T]
```

to

``` scala
def create[T](f: Subscriber[T] => Unit): Observable[T]
```

would be very dangerous, because code written for the old `create` would still compile, but do something different.

``` scala
val myObservable = Observable.create[Int](observer => {
  ...
  Subscription {
    // unsubscription handler
  }
})
```

Such code would still compile with the new `create`, because `Subscriber <: Observer` and function args are contravariant, and for the return type, by discarding the value of the last expression, we get `Unit`. But unsubscribing from `myObservable` would have no effect, because this code does not check `isUnsubscribed` and and the `Subscription` it returns would be ignored.

**Comment by experquisite** [comment]
Second that, it would be better to break builds IMO.

**Comment by headinthebox** [comment]
That's not what I propose ... at all :-). Just to replace the _name_ `s` with `f`. Hence the subject "bike shedding"

**Comment by samuelgruetter** [comment]
Oh I see, so I'm glad that you didn't try to sell us the removal of the nuclear reactor's "shut down" button as "bike shedding" ;-)

**Comment by zsxwing** [comment]
@headinthebox I checked these two methods. Now they are:

```
def create[T](func: Observer[T] => Subscription): Observable[T]
def apply[T](f: Subscriber[T] => Unit): Observable[T]
```

So do you mean changing `func` to `f`?

**Comment by headinthebox** [comment]
Whatever the scala ways is, which is `f` I believe.

**Comment by zsxwing** [comment]
Fixed in #92
